### DISCUSSION TITLE
theme preview: add more components

### DISCUSSION THREAD

**Discussion by Angelk90** [zed-industries/zed#24234] [Zed GPUI]
### Check for existing issues

- [X] Completed

### Describe the feature

It would be nice to be able to add other components in particular input:

- [ ] single line editor
- [ ] multiline editor

<details><summary>screenshots</summary>

<img width="1090" alt="Screenshot 2024-12-04 alle 15 06 19" src="https://github.com/user-attachments/assets/14f4f6ef-4bbd-425a-ac6e-a31f82e2d3ff">
<img width="1083" alt="Screenshot 2024-12-04 alle 15 06 45" src="https://github.com/user-attachments/assets/fccc9f28-0696-480a-b16f-165345143a0f">

</details>

### Environment

```
Zed: v0.164.0 (Zed Preview)
OS: macOS 14.1.1
Memory: 16 GiB
Architecture: x86_64
```

**Comment by notpeter** [comment]
gpui does not have date/time picker component so that's very much out-of-scope and I've removed it from your description.

But it would be a nice enhancement to display additional components in the theme preview. For example:
- [ ] Single-line Editor
- [ ] Multi-line Editor
- [ ] Multi-line Editor (fixed height)
- [ ] Editor components:
  - [ ] Indent Guides
  - [ ] Inlay Hints
  - [ ] Scrollbar
  - [ ] Breadcrumbs
  - [ ] Popover
- [ ] Tab bar
- [ ] Icons (selected/active, inactive)
- [ ] Toast notifications (temporary error)
- [ ] Context Menus (e.g. right click)
- [ ] Action Menus (e.g. `cmd-.`)

Not sure how much work it would be to include these in the Theme Preview (e.g. statically rendering a right-click menu into another surface may require a non-trivial amount of work for minimal benefit).
